update/v1alpha1/types_update_status.go

```diff
@@ -0,0 +1,394 @@
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// UpdateStatus reports status for in-progress cluster version updates
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:level=4
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:path=updatestatuses,scope=Cluster
+// +openshift:api-approved.openshift.io=https://github.com/openshift/api/pull/2012
+// +openshift:file-pattern=cvoRunLevel=0000_00,operatorName=cluster-version-operator,operatorOrdering=02
+// +openshift:enable:FeatureGate=UpgradeStatus
+// +kubebuilder:metadata:annotations="description=Provides health and status information about OpenShift cluster updates."
+// +kubebuilder:metadata:annotations="displayName=UpdateStatuses"
+type UpdateStatus struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec is empty for now, UpdateStatus is purely status-reporting API. In the future spec may be used to hold
+	// configuration to drive what information is surfaced and how
+	// +required
+	Spec UpdateStatusSpec `json:"spec"`
+	// status exposes the health and status of the ongoing cluster update
+	// +optional
+	Status UpdateStatusStatus `json:"status"`
+}
+
+// UpdateStatusSpec is empty for now, UpdateStatus is purely status-reporting API. In the future spec may be used
+// to hold configuration to drive what information is surfaced and how
+type UpdateStatusSpec struct {
+}
+
+// +k8s:deepcopy-gen=true
+
+// UpdateStatusStatus is the API about in-progress updates. It aggregates and summarizes UpdateInsights produced by
+// update informers
+type UpdateStatusStatus struct {
+	// conditions provide details about the controller operational matters
+	// +listType=map
+	// +listMapKey=type
+	// +patchStrategy=merge
+	// +patchMergeKey=type
+	// +optional
+	// +kubebuilder:validation:MaxItems=10
+	Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+
+	// controlPlane contains a summary and insights related to the control plane update
+	// +required
```

Comment:
All top level status fields should be optional. The lack of omitempty on a parent status could lead to odd minimal JSON marshalling and breaking required field requirements further down

Reply:
I assume it is enough to mark this as optional, and not do anything else like adding `omitempty` tags somewhere (bc it does not do anything for structs afaik?) or make some members pointers or something.

Reply:
If the struct contains required fields, it must be marked omitempty and be a pointer

Reply:
Is it feasible to have `Status` resource itself an `omitempty` pointer? So a resource may not have a status, but if it does, it has to have `controlPlane`?

Reply:
Yes, you could do that, but conventions have generally suggested not to and that there should be no required top level status fields. I'll let you make the call here

Reply:
For now I'll go with the conventions, clients will need to handle the absence gracefully